app/gui/src/dashboard/utilities/vue.ts

```diff
@@ -0,0 +1,23 @@
+/** @file Helpers for using Vue components from React. */
+import * as React from 'react'
+
+import { applyPureVueInReact } from 'veaury'
+import type { AllowedComponentProps, VNodeProps } from 'vue'
+import type { ComponentProps } from 'vue-component-type-helpers'
+
+/** Extracts the properties defined by a component, excluding various Vue internals. */
+type VueComponentProps<T> = Omit<ComponentProps<T>, keyof AllowedComponentProps | keyof VNodeProps>
+
+/** Creates a React component wrapping a Vue component. */
+export function vueComponent<T extends { default?: unknown }>(
+  lazyImport: () => Promise<T>,
+): React.LazyExoticComponent<(props: VueComponentProps<T['default']>) => React.JSX.Element> {
+  return React.lazy(async () => {
```

Comment:
Lazy shouldn't be mandatory, bacause it's a perf optimization and usage depends on the context.